pkg/graphql/config_factory_federation.go

```diff
@@ -0,0 +1,135 @@
+package graphql
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/jensneuse/graphql-go-tools/pkg/astparser"
+	graphqlDataSource "github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/graphql_datasource"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/plan"
+	"github.com/jensneuse/graphql-go-tools/pkg/federation"
+)
+
+type federationEngineConfigFactoryOptions struct {
+	httpClient *http.Client
+}
+
+type FederationEngineConfigFactoryOption func(options *federationEngineConfigFactoryOptions)
+
+func WithFederationHttpClient(client *http.Client) FederationEngineConfigFactoryOption {
+	return func(options *federationEngineConfigFactoryOptions) {
+		options.httpClient = client
+	}
+}
+
+func NewFederationEngineConfigFactory(dataSourceConfigs []graphqlDataSource.Configuration, opts ...FederationEngineConfigFactoryOption) *FederationEngineConfigFactory {
+	options := federationEngineConfigFactoryOptions{
+		httpClient: &http.Client{
+			Timeout: time.Second * 10,
+			Transport: &http.Transport{
+				MaxIdleConnsPerHost: 1024,
+				TLSHandshakeTimeout: 0 * time.Second,
+			},
+		},
+	}
+
+	for _, optFunc := range opts {
+		optFunc(&options)
+	}
+
+	return &FederationEngineConfigFactory{
+		httpClient:        options.httpClient,
+		dataSourceConfigs: dataSourceConfigs,
+	}
+}
+
+type FederationEngineConfigFactory struct {
+	httpClient        *http.Client
+	dataSourceConfigs []graphqlDataSource.Configuration
+	schema            *Schema
+}
+
+func (f *FederationEngineConfigFactory) SetMergedSchemaFromString(mergedSchema string) (err error) {
+	f.schema, err = NewSchemaFromString(mergedSchema)
+	if err != nil {
+		return fmt.Errorf("set merged schema: %s", err.Error())
```

Comment:
should we annotate errors with info on the origin? e.g. a reference to the FederationEngineConfigFactory